src/app/services/slices/history.slice.ts

```diff
@@ -0,0 +1,33 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { IHistoryItem } from '../../../types/history';
+
+const initialState: IHistoryItem[] = [];
+
+const historySlice = createSlice({
+  name: 'history',
+  initialState,
+  reducers: {
+    addHistoryItemSuccess(state, action: PayloadAction<IHistoryItem>) {
+      state.push(action.payload);
+    },
+    bulkAddHistoryItemsSuccess(state, action: PayloadAction<IHistoryItem[]>) {
+      state.push(...action.payload);
+    },
+    removeHistoryItemSuccess(state, action: PayloadAction<IHistoryItem>) {
+      state = state.filter(item => item !== action.payload);
+    },
+    removeAllHistoryItemsSuccess(state, action: PayloadAction<string[]>) {
+      state = state.filter(item => !action.payload.includes(item.createdAt));
+    }
```

Comment:
I was reading this https://redux-toolkit.js.org/usage/immer-reducers#resetting-and-replacing-state in regards to changing the entire state. In our case, we are creating a new array with `filter`, so to mutate the state, we need to return this filtered array, and not assign it to the `state` variable.